Update dependency to implement horizon shading feature

thomay86 opened the pull request:
The horizon profile can be specified in a text file and the forecast is calculated accordingly.

The requirements field in manifest.json can be updated, once [the pull request in the dependency](https://github.com/rany2/open-meteo-solar-forecast/pull/37#issue-3671033272) has been merged and published as a (pre-)release.

thomay86 committed:
works now with pre-release v0.1.0-beta2 to implement the horizon shading feature

thomay86 committed:
Document horizon shading feature in README

Added information about horizon shading and its configuration.

thomay86 committed:
corrections to const.py and de/en.json

rany2 commented:
This looks good to me, I just have one small nitpick. For HACS users, it would be better for horizon.txt file to be outside of custom_components. Perhaps we can use horizon.txt inside custom_components as the fallback path, but default to a location inside /config directly.

rany2 commented:
Also hardcoding it as /config is not ideal as during development/for non-docker or HAOS users; that is unlikely to be the HA path.

thomay86 committed:
Update manifest.json

removed suncalc requirement as this will be specified in open-meteo-solar-forecast pyproject.toml

rany2 commented:
> How about letting the user specify the file location as a string in the config flow. Would also be more flexible to be able to use multiple files for multiple instances, if needed.

That seems fine too. Another option is to just allow the user to paste their horizon file in the config flow.

thomay86 committed:
made the path to the horizon file editable as a string user input
logs a 'blocking call' warning in HA, but runs as expected

rany2 commented on custom_components/open_meteo_solar_forecast/coordinator.py:
```diff
@@ -42,6 +46,10 @@ def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
         ac_kwp = ac_kwp / 1000 if ac_kwp else None
+        
+        # Read horizon data from text file
+        horizon_data = numpy.genfromtxt(entry.options.get(CONF_HORIZON_FILEPATH) , delimiter="\t", dtype=float)
```
Hmm, I'll try to see if it can be worked around.

thomay86 committed:
adapted for partial shading
specified suncalc in manifest again because of downstream issues with numpy specification in poetry

rany2 commented:
@thomay86 Should be fixed.

thomay86 committed:
update manifest for v0.1.0-beta5

Newspaperman57 commented:
As a point-of-reference, i've just installed this PR since dampenings did not model my production correctly, and it seems to work as expected :+1: 

Personally i would have preferred just pasting the contents of the file directly into the config UI but to each his own, this worked just fine too.

I did have one minor issue:
My first attempt had a wrong syntax for the horizon-file (my fault entirely) and the integration failed to load, but it was not immediately obvious the horizons file was the problem.

The logs contained (i think) a ValueError in one case and a "object not subscriptable" in another, and my coding-background quickly pieced together that it was caused by the horizonsfile, but i'm not sure an average user would make that connection

thomay86 committed:
Update README with partial shading information

Added information about treating shading as partial in horizon settings.

rany2 commented:
@ivankravets open-meteo-solar-forecast 0.1.23 has been published with these changes :)

thomay86 commented:
@rany2 thanks! I'll try to update the HA repo next week.

thomay86 committed:
update manifest.json to point to resource open-meteo-solar-forecast==0.1.23
added checking routine for horizon file

thomay86 committed:
Enhance horizon shading documentation in README

Updated horizon shading section for clarity and detail.

thomay86 committed:
added string and DE translation for partial_shading

thomay86 committed:
correction of last commit

thomay86 commented:
Horizon implementation updated to OMSF v0.1.23 and tested!
